### PR TITLE
Implemented password salting

### PR DESCRIPTION
**Pull request by Ilithor**

### CODE REVIEW

File: server/routes/user.js [hunk 1]
```diff
@@ -8,9 +8,10 @@ import {
   getAuthenticatedUser,
   getUserDetail,
   fetchLikeList,
+  saltExistingPassword,
```
Comment (jeffersoneagley) [inline server/routes/user.js]: delet

File: server/services/user.service.js [hunk 2]
```diff
@@ -60,10 +64,12 @@ export const register = async userParam => {
   if (invalid) {
     return Promise.reject(invalid);
   }
+  // Salt and hash password
+  user.credential.password = bcrypt.hashSync(userParam.password, 10);
```
Comment (jeffersoneagley) [inline server/services/user.service.js]: Shouldn't this be done client-side?

Comment (Ilithor) [reply server/services/user.service.js]: fixed

File: server/routes/user.js (hunk 1 above)
Comment (Ilithor) [reply server/routes/user.js]: fixed

File: src/components/login/loginContext.js [hunk 3]
```diff
@@ -65,6 +72,19 @@ export const LoginProvider = ({ children }) => {
     }
   };
 
+  /** Hashes provided password
+   *
+   * @param {string} password
+   * @returns {string | Error}
+   */
+  const hashPassword = async password => {
```
Comment (jeffersoneagley) [inline src/components/login/loginContext.js]: We're still doing the hashing serverside?

Comment (Ilithor) [reply src/components/login/loginContext.js]: What I decided to do was hash it client side, then rehash again and then salt. You can never hash a password too many times.

Comment (jeffersoneagley) [reply src/components/login/loginContext.js]: Beautiful 🔐🔐🔐

**Pull request merged**